include/ensmallen_bits/spsa/spsa_impl.hpp

```diff
@@ -98,8 +100,12 @@ inline double SPSA::Optimize(
     overallObjective = function.Evaluate(iterate);
   }
 
-  // Calculate final objective.
-  return function.Evaluate(iterate);
+  // Calculate final objective if exactObjective is set to true.
+  if (exactObjective)
+  {
+    overallObjective = function.Evaluate(iterate);
+  }
+  return overallObjective;
```

Comment:
SPSA doesn't work on decomposable functions, so we can avoid the change here too.